resources/karaf.rb

```diff
@@ -108,7 +98,7 @@ def client_command
   end
 
   # Create the service
-  if node['platform'] == 'centos' and node['platform_version'] >= '7'
+  if ::File.exists?("#{karaf_path}/bin/karaf.service") and node['platform'] == 'centos' and node['platform_version'] >= '7'
```

Comment:
I think the File.exists check is preventing the service to be created on Centos 7; I think the file may not exist when this code block is hit. At least, that's what I'm seeing locally.